test/compilable/betterCarray.d

```diff
@@ -0,0 +1,17 @@
+/* REQUIRED_ARGS: -betterC
+   PERMUTE_ARGS:
+*/
+
+import core.stdc.stdio;
+
+extern (C) int main(char** argv, int argc) {
+    printf("hello world\n");
+    int[3] a;
+    foo(a[], 3);
+    return 0;
+}
+
+int foo(int[] a, int i)
+{
+    return a[i];
```

Comment:
Not testing much here. We're not even linking the compilable tests.
You should at least run this and grep in the output, at least on some platforms.